internal/cmd/gocdk/fswatch.go

```diff
@@ -0,0 +1,205 @@
+// Copyright 2019 The Go Cloud Development Kit Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sort"
+	"time"
+
+	"golang.org/x/xerrors"
+)
+
+// waitForDirChange blocks until the directory changes or the context's Done
+// channel is closed.
+func waitForDirChange(ctx context.Context, path string, pollInterval time.Duration) error {
+	tick := time.NewTicker(pollInterval)
+	defer tick.Stop()
+	init, err := snapshotTree(path)
```

Comment:
Nit: consider calling this `snapshot` instead of `init`? Or `initSnapshot`?

Reply:
Done.